base_station/gui/src/components/ToggleButton.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+    <label :for="id + '_button'" :class="{'active': currentState}" class="toggle__button">
+        <span v-if="currentState" class="toggle__label">{{ labelEnableText }}</span>
+        <span v-if="! currentState" class="toggle__label">{{ labelDisableText }}</span>
+
+        <input type="checkbox" :id="id + '_button'" v-model="checkedValue">
+        <span class="toggle__switch"></span>
+    </label>
+</template>
+
+<script>
+export default {
+    data() {
+        return {
+            currentState: this.currentState
```

Comment:
Seems weird cause this would make it so this.currentState is a bit ambiguous. Is it pointing to the member of data() or props? I think this can be removed altogether though I'm not 100$ sure